en/mavgen_c/README.md

```diff
@@ -59,16 +60,164 @@ When compiling the project, we recommend that you specify the top-level output d
 $ gcc ... -I generated/include -I generated/include/common ...
 ```
 
-## Multiple Streams ("channels") {#streams}
+## Multiple Streams ("channels") {#channels}
 
-The C MAVLink library utilizes a "channel" metaphor to allow for simultaneous processing of multiple, independent MAVLink streams in the same program. 
-It is therefore important to use the correct channel for each operation as all receiving and transmitting functions provided by MAVLink require a channel.
+The C MAVLink library utilizes a "channel" metaphor to allow for simultaneous processing of multiple, independent MAVLink streams in the same program.
+All receiving and transmitting functions provided by this library require a channel, and it is important to use the correct channel for each operation.
 
+Up to 4 channels can be defined on MCU/ARM7 while up to 16 may be defined on Linux/Windows.
 If only one MAVLink stream exists, channel 0 should be used by specifying the `MAVLINK_COMM_0` constant.
 
+
 ## Receiving
 
-MAVLink reception is then done using `mavlink_helpers.h:mavlink_parse_char()`.
+MAVLink reception/decoding is done in a number of phases: 
+1. Parse the incoming stream into objects representing each packet (`mavlink_message_t`).
+1. Decode the MAVLink message contained in the packet payload into a C struct (that has fields mapping the original XML definition).
```

Comment:
Sequence error?

Reply:
No, markdown uses `1.` to indicate any numbered bullet, and provides autonumbering.